tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js

```diff
@@ -1,25 +1,283 @@
+      const daySelectorXpath = util.format(this.elements.dateTimeDayPicker.selector, day)
+      return {
+        selector: daySelectorXpath,
+        locateStrategy: 'xpath'
+      }
+    },
+    /**
+     * sets provided year in expiry date field on webUI
+     *
+     * @param selector selector for expiry date "year"
+     * @returns {Promise<void>}
+     */
+    setExpiryDateYear: async function (selector) {
+      await this
+        .waitForElementVisible('@dateTimePopupYear')
+        .waitForAnimationToFinish()
+        .waitForElementEnabled(
+          this.elements.dateTimePopupYear.selector
+        )
+        .click('@dateTimePopupYear')
+        .waitForElementVisible(selector)
+        .click(selector)
+        .click('@dateTimeOkButton')
+    },
+    /**
+     * sets provided month in expiry date field on webUI
+     *
+     * @param selector selector for expiry date "month"
+     * @returns {Promise<void>}
+     */
+    setExpiryDateMonth: async function (selector) {
+      await this
+        .waitForElementVisible('@dateTimePopupDate')
+        .click('@dateTimePopupDate')
+        .waitForElementVisible(selector)
+        .click(selector)
+        .click('@dateTimeOkButton')
+    },
+    /**
+     * sets provided day in expiry date field on webUI
+     *
+     * @param selector selector for expiry date "day"
+     * @returns {Promise<void>}
+     */
+    setExpiryDateDay: async function (selector) {
+      await this
+        .waitForElementVisible(selector)
+        .click(selector)
+        .click('@dateTimeOkButton')
+    },
+    setPublicLinkRole: async function (value) {
+      const roleSelectorXpath = util.format(this.elements.roleButton.selector, value)
+      const roleSelector = { locateStrategy: 'xpath', selector: roleSelectorXpath }
+      await this
+        .waitForElementVisible(roleSelector)
+        .click(roleSelector)
+    },
+    setPublicLinkName: async function (value) {
+      await this
+        .waitForElementVisible('@publicLinkNameInputField')
+        .clearValue('@publicLinkNameInputField')
+        .setValue('@publicLinkNameInputField', value)
+    },
+    setPublicLinkPassword: async function (value) {
+      await this
+        .waitForElementVisible('@publicLinkPasswordField')
+        .clearValue('@publicLinkPasswordField')
+        .setValue('@publicLinkPasswordField', value)
+    },
+    setPublicLinkExpiryDate: async function (value) {
+      const date = value.split(/-/)
+      const yearSelector = this.setExpiryDateYearSelectorXpath(date[0])
+      const monthSelector = this.setExpiryDateMonthSelectorXpath(date[1])
+      const daySelector = this.setExpiryDateDaySelectorXpath(date[2])
+      await this
+        .initAjaxCounters()
+        .waitForElementVisible('@linkExpirationDateField')
+        .click('@linkExpirationDateField')
+      await this.setExpiryDateYear(yearSelector)
+      await this.setExpiryDateMonth(monthSelector)
+      await this.setExpiryDateDay(daySelector)
```

Comment:
Nit:
1. Remove `async` and just chain everything and return.
2. Don't send the selectors in, just send the appropriate value.